jujugui/static/gui/src/app/app.js

```diff
@@ -1158,39 +1152,15 @@ YUI.add('juju-gui', function(Y) {
           // reused for handling high availability controllers.
           var addresses = envData['host-ports'];
           var wssData = addresses[addresses.length - 1].split(':');
```

Comment:
It feels a little weird that the code from 1148-1154 are here and not in the `createSocketURL` method. That would make the call signature `this.createSocketURL(wssData, envData)` and then the extraction inside there. I don't feel super strongly about this but  I think it'll help clean this up even more than you already are.